metals/src/main/scala/scala/meta/internal/decorations/SemanticdbTreePrinter.scala

```diff
@@ -134,24 +138,33 @@ class SemanticdbTreePrinter(
       case s.StringConstant(value) => value
     }
 
-  def printTree(t: s.Tree): Option[String] =
+  def printTree(
+      t: s.Tree,
+      isExplicitTuple: => Boolean = false
+  ): Option[String] =
     t match {
       case s.Tree.Empty => None
       case s.OriginalTree(_) => None
-      case s.TypeApplyTree(function, typeArguments) =>
-        Some(printTree(function).getOrElse("") + printTypeArgs(typeArguments))
+      case s.TypeApplyTree(function, typeArguments)
+          if shouldPrintTypeApply(function) || isExplicitTuple =>
```

Comment:
Maybe reorganize/rename these functions?

```suggestion
          if isNotTupleLiteral(function, range) =>
          
     ...
     def isNotTupleLiteral(t: s.Tree, range: Range): Boolean = t match {
        case SelectTree(_, Some(id)) => isTuple(id.symbol) && !isExplicitTuple(range)
        case _ => true
     }
```

Reply:
I changed the name, kept the `isExplicitTuple` parameter, since we don't have access to text at that point.